textile/chat-features.textile

```diff
@@ -9,5 +9,495 @@ jump_to:
+in whichever format is most idiomatic to the platform.
+
+h3(#chat-structs-message). Messages
+
+<pre>
+  {
+    "timeserial": "cbfqxperABgItU52203559@1726232498871-0",
+    "roomId": "my-room",
+    "clientId": "who-sent-the-message",
+    "text": "my-message",
+    "createdAt": DateTime(),
+    "metadata": {
+      "foo": {
+        "bar": 1
+      }
+    },
+    "headers": {
+      "baz": "qux"
+    }
+  }
+</pre>
+
+Determining the global order of messages may be achieved by comparing the timeserials. See @CHA-M2@ for more information.
+
+h3(#chat-structs-ephemeral-reactions). Ephemeral Room Reactions
+
+<pre>
+  {
+    "type": ":heart:",
+    "roomId": "my-room",
+    "clientId": "who-sent-the-message",
+    "createdAt": DateTime(),
+    "metadata": {
+      "foo": {
+        "bar": 1
+      }
+    },
+    "headers": {
+      "baz": "qux"
+    }
+  }
+</pre>
+
+h3(#chat-structs-presence-member). Presence Member
+
+<pre>
+  {
+    "clientId": "who-sent-the-message",
+    "action": "enter",
+    "updatedAt": DateTime(),
+    "data": {}, // Whatever the user-provided data is. It must be destructured from the internal format we use.
+    "extras": {
+      "headers": {
+        "baz": "qux"
+      }
+    }
+  }
+</pre>
+
+h3(#chat-structs-presence-event). Presence Event
+
+<pre>
+  {
+    "clientId": "who-is-in-presence",
+    "action": "enter",
+    "timestamp": DateTime(),
+    "data": {}, // Whatever the user-provided data is. It must be destructured from the internal format we use.
+  }
+</pre>
+
+h2(#error-codes). Chat-specific Error Codes
+
+This section contains error codes that are specific to Chat. If a specific error code is not listed for a given circumstance, the most appropriate general error code shall be used. For example @400xx@ for client errors or @500xx@ for server errors.
+
+<pre>
+  // The messages feature failed to attach.
+  MessagesAttachmentFailed = 102001
+
+  // The presence feature failed to attach.
+  PresenceAttachmentFailed = 102002
+
+  // The reactions feature failed to attach.
+  ReactionsAttachmentFailed = 102003
+
+  // The occupancy feature failed to attach.
+  OccupancyAttachmentFailed = 102004
+
+  // The typing feature failed to attach.
+  TypingAttachmentFailed = 102005
+  // 102006 - 102049 reserved for future use for attachment errors
+
+  // The messages feature failed to detach.
+  MessagesDetachmentFailed = 102050
+
+  // The presence feature failed to detach.
+  PresenceDetachmentFailed = 102051
+
+  // The reactions feature failed to detach.
+  ReactionsDetachmentFailed = 102052
+
+  // The occupancy feature failed to detach.
+  OccupancyDetachmentFailed = 102053
+
+  // The typing feature failed to detach.
+  TypingDetachmentFailed = 102054
+  // 102055 - 102099 reserved for future use for detachment errors
+
+  // The room has experienced a discontinuity.
+  RoomDiscontinuity = 102100
+
+  // Unable to perform operation;
+
+  // Cannot perform operation because the room is in a failed state.
```

Comment:
Given your [comment](https://github.com/ably-labs/ably-chat-swift/issues/73#issuecomment-2385770039) in another issue about wanting to reserve the term "state" for a larger, composite state, would it be more appropriate that the error name and message refer to "status" instead of "state"?

Reply:
Actually, looking at this, should this be called `RoomIsFailed` for consistency with `RoomIsReleasing` and `RoomIsReleased`? (and question about error messages stands)

Reply:
Perhaps - though I might defer this one for later if that's ok!

Reply:
Sure 👍